src/main/kotlin/com/terraformation/backend/accelerator/ModuleEventService.kt

```diff
@@ -73,4 +95,22 @@ class ModuleEventNotifier(
       }
     }
   }
+
+  fun updateEventStatusIfEventUpToDate(event: ModuleEventScheduledEvent, status: EventStatus) {
+    systemUser.run {
+      val moduleEvent =
+          try {
+            eventStore.fetchOneById(event.eventId)
+          } catch (e: EventNotFoundException) {
+            log.error("Module event ${event.eventId} not found.")
+            return@run
+          }
+
+      if (moduleEvent.revision == event.revision) {
+        eventStore.updateEventStatus(event.eventId, status)
+      } else {
+        log.info("Module event ${event.eventId} has been changed. Not updating status.")
+      }
```

Comment:
It'd be good to put this in the `try` block for a couple reasons:

- Won't need the `return@run` statement.
- If the update throws an exception, it'll be caught here rather than bubbling up to the job scheduler, which would cause the job to be retried several times.